crates/accelerate/src/gate_direction.rs

```diff
@@ -142,8 +157,505 @@ where
+            continue;
+        };
+
+        // Take into account qubit index mapping if we're inside a control-flow block
+        let (op_args0, op_args1) = if let Some(mapping) = qubit_mapping {
+            (mapping[op_args[0].index()], mapping[op_args[1].index()])
+        } else {
+            (op_args[0], op_args[1])
+        };
+
+        if gate_complies(packed_inst, &[op_args0, op_args1]) {
+            continue;
+        }
+
+        // If the op has a pre-defined replacement - replace if the other direction is supported otherwise error
+        // If no pre-defined replacement for the op - if the other direction is supported error saying no pre-defined rule otherwise error saying op is not supported
+        if let OperationRef::Standard(std_gate) = packed_inst.op.view() {
+            match std_gate {
+                StandardGate::CXGate
+                | StandardGate::ECRGate
+                | StandardGate::CZGate
+                | StandardGate::SwapGate
+                | StandardGate::RXXGate
+                | StandardGate::RYYGate
+                | StandardGate::RZZGate
+                | StandardGate::RZXGate => {
+                    if gate_complies(packed_inst, &[op_args1, op_args0]) {
+                        // Store this for replacement outside the dag.op_nodes loop
+                        nodes_to_replace.push((node, replace_dag(py, std_gate, packed_inst)?));
+                        continue;
+                    } else {
+                        return Err(TranspilerError::new_err(format!(
+                            "The circuit requires a connection between physical qubits {:?} for {}",
+                            op_args,
+                            packed_inst.op.name()
+                        )));
+                    }
+                }
+                _ => {}
+            }
+        }
+        // No matching replacement found
+        if gate_complies(packed_inst, &[op_args1, op_args0])
+            || has_calibration_for_op_node(py, dag, packed_inst, &[op_args1, op_args0])?
+        {
+            return Err(TranspilerError::new_err(format!("{} would be supported on {:?} if the direction was swapped, but no rules are known to do that. {:?} can be automatically flipped.", packed_inst.op.name(), op_args, vec!["cx", "cz", "ecr", "swap", "rzx", "rxx", "ryy", "rzz"])));
+            // NOTE: Make sure to update the list of the supported gates if adding more replacements
+        } else {
+            return Err(TranspilerError::new_err(format!(
+                "{} with parameters {:?} is not supported on qubits {:?} in either direction.",
+                packed_inst.op.name(),
+                packed_inst.params_view(),
+                op_args
+            )));
+        }
+    }
+
+    for (node, op_blocks) in ops_to_replace {
+        let NodeType::Operation(packed_inst) = &dag.dag()[node] else {
+            panic!("PackedInstruction is expected");
+        };
```

Comment:
Use `unwrap_operation()` to extract a known op_node.
```suggestion
        let packed_inst: &PackedInstruction = &dag.dag()[node].unwrap_operation();
```

Reply:
Ah, cool! Done in 4b363a19bd5920926d3f31748738f7b2070e4dd2